### ISSUE TITLE
Unable to Compile Master: Call of Overloaded Function Is Ambiguous

### ISSUE THREAD

**Issue by nandeeka** [tensor-compiler/taco#380]
I just cloned the master branch of this repository and was trying to install it according to the README. However, I get the following error when trying to compile `tests-workspaces.cpp.o` during `make -j8`. Please let me know if any other information would be helpful.

```
[ 69%] Building CXX object test/CMakeFiles/taco-test.dir/tests-workspaces.cpp.o
/home/ndnayak2/src/taco/test/tests-workspaces.cpp: In member function ‘virtual void workspaces_tile_vecElemMul_NoTail_Test::TestBody()’:
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:15:38: error: call of overloaded ‘Tensor(const char [2], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> A("A", {16}, {Dense});
                                      ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::_
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = st
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:16:38: error: call of overloaded ‘Tensor(const char [2], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> B("B", {16}, {Dense});
                                      ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::_
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = st
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:17:38: error: call of overloaded ‘Tensor(const char [2], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> C("C", {16}, {Dense});
                                      ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::_
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = st
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:45:52: error: call of overloaded ‘Tensor(const char [9], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> expected("expected", {16}, {Dense});
                                                    ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::_
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = s$
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp: In member function ‘virtual void workspaces_tile_vecElemMul_Tail1_Test::TestBody()’:
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:55:38: error: call of overloaded ‘Tensor(const char [2], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> A("A", {16}, {Dense});
                                      ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::$
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = s$
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:56:38: error: call of overloaded ‘Tensor(const char [2], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> B("B", {16}, {Dense});
                                      ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::$
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = s$
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:57:38: error: call of overloaded ‘Tensor(const char [2], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> C("C", {16}, {Dense});
                                      ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::$
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = s$
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:83:52: error: call of overloaded ‘Tensor(const char [9], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> expected("expected", {16}, {Dense});
                                                    ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::_
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = st
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp: In member function ‘virtual void workspaces_tile_vecElemMul_Tail2_Test::TestBody()’:
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:93:38: error: call of overloaded ‘Tensor(const char [2], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> A("A", {17}, {Dense});
                                      ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::_
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = st
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:94:38: error: call of overloaded ‘Tensor(const char [2], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> B("B", {17}, {Dense});
                                      ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::_
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = st
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:95:38: error: call of overloaded ‘Tensor(const char [2], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> C("C", {17}, {Dense});
                                      ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::_
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = st
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:121:52: error: call of overloaded ‘Tensor(const char [9], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> expected("expected", {17}, {Dense});
                                                    ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::_
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = st
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp: In member function ‘virtual void workspaces_tile_denseMatMul_Test::TestBody()’:
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:142:38: error: call of overloaded ‘Tensor(const char [2], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> A("A", {16}, {Dense});
                                      ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::_
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = st
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:143:38: error: call of overloaded ‘Tensor(const char [2], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> B("B", {16}, {Dense});
                                      ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::_
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = st
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:144:38: error: call of overloaded ‘Tensor(const char [2], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> C("C", {16}, {Dense});
                                      ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::_
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = st
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
/home/ndnayak2/src/taco/test/tests-workspaces.cpp:170:52: error: call of overloaded ‘Tensor(const char [9], <brace-enclosed initializer list>, <brace-enclosed initializer list>)’ is ambiguous
   Tensor<double> expected("expected", {16}, {Dense});
                                                    ^
In file included from /home/ndnayak2/src/taco/test/test.h:11:0,
                 from /home/ndnayak2/src/taco/test/tests-workspaces.cpp:4:
/home/ndnayak2/src/taco/include/taco/tensor.h:1050:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::Format) [with CType = double; std::__cxx11::string = std::_
_cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, Format format)
 ^
/home/ndnayak2/src/taco/include/taco/tensor.h:1045:1: note: candidate: taco::Tensor<CType>::Tensor(std::__cxx11::string, std::vector<int>, taco::ModeFormat) [with CType = double; std::__cxx11::string = st
d::__cxx11::basic_string<char>]
 Tensor<CType>::Tensor(std::string name, std::vector<int> dimensions, 
 ^
test/CMakeFiles/taco-test.dir/build.make:806: recipe for target 'test/CMakeFiles/taco-test.dir/tests-workspaces.cpp.o' failed
make[2]: *** [test/CMakeFiles/taco-test.dir/tests-workspaces.cpp.o] Error 1
CMakeFiles/Makefile2:212: recipe for target 'test/CMakeFiles/taco-test.dir/all' failed
make[1]: *** [test/CMakeFiles/taco-test.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by xysmlx** [comment]
Get the same error. I am using ubuntu-16.04, cmake-3.18.0 and gcc-5.4.0.

**Comment by xysmlx** [comment]
I can successfully build the master branch with Ubuntu-18.04, cmake-3.19.1 and gcc-7.5.0.

**Comment by Infinoid** [comment]
You're right, I ran ubuntu 16.04 in a docker container and I'm able to see the same build failures.

I've fixed some similar problems in #340 in the past, but this test is new.  I just posted a PR to fix it.

Thanks!
